content/articles/using-machine-learning-to-translate-brain-signals-to-text/index.md

```diff
@@ -0,0 +1,136 @@
+
+![t-SNE](/engineering-education/using-machine-learning-to-translate-brain-signals-to-text/t-sne.PNG)
+
+*[Image Source: Nature](https://www.nature.com/articles/s41586-021-03506-2)*
+
+The figure represents the 2D visualization of the neural activity made using t-SNE. 
+
+### Decoding handwritten sentences
+
+So, now that they were successful in representing neural activities in their respective letters, the researchers faced a new challenge. Could they replicate the same success in decoding entire handwritten sentences instead of only letters in real-time?
+
+Let's see how they went about with this.
+
+They achieved this by using the Recurrent Neural Network (RNN). You can read more about RNN's [here](https://en.wikipedia.org/wiki/Recurrent_neural_network).
+
+The RNN is trained to convert the brain's neural activity data into probabilities. These probabilities describe the likelihood of each character being written within a certain window of time. It assigns the probability of that data being associated with any particular character or sign based on the embedding that we explained earlier. 
+
+They used a set of 31 characters. These characters consisted of 26 lower-case letters from the alphabet, a question mark (?), comma (,), apostrophe('), full stop (-), and space (>)
+
+They performed this in two different ways. Using an online and offline system.
+
+![The online and offline system](/engineering-education/using-machine-learning-to-translate-brain-signals-to-text/online-offline-system.PNG)
+
+*[Image Source: Nature](https://www.nature.com/articles/s41586-021-03506-2)*
+
+In the online system, they used a threshold system. If the probability of a certain set of data with letters was high enough, then it would be assigned that letter. The assigned letters would then be used in the sentence.
+In the offline system, they used a large-vocabulary language model to simulate an auto-correct feature. It fixed errors by predicting the probability that a letter came next in a sequence of letters. 
+
+### Results
+
+This research shows results for 5 days from one study participant who already had a system implanted in his brain.
+
+To avoid bombarding you with a lot of jargon, I've summarized the results of the finding for you:
+
+1. As the patient continued to do the task of imagining letters and allowing the computer to process these signals repeatedly, the patient was able to type 90 characters per minute for each trial day. This doubled the record compared to previous BCI records for typing. 
+
+2. Since our brain's baseline neural signals change day after day, the handwriting decoder had to be retrained with new data every single day. This retraining had to be accounted for especially when working with models like this. You wouldn't want your training signals to look this way one day and a different way another day. This would make your model not work correctly.
```

Comment:
```suggestion
2. Since our brain's baseline neural signals change day after day, the handwriting decoder had to be retrained with new data every single day. This retraining had to be accounted for especially when working with models like this. You wouldn't want your training signals to look a certain way one day and a different way another day. This would make your model work incorrectly.
```